server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxChangesMethod.scala

```diff
@@ -0,0 +1,89 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.method
+
+import eu.timepit.refined.auto._
+import javax.inject.Inject
+import org.apache.james.jmap.api.change.{MailboxChange, MailboxChangeRepository}
+import org.apache.james.jmap.api.model.{AccountId => JavaAccountId}
+import org.apache.james.jmap.core.CapabilityIdentifier.{CapabilityIdentifier, JMAP_MAIL}
+import org.apache.james.jmap.core.Invocation.{Arguments, MethodName}
+import org.apache.james.jmap.core.{Invocation, Properties, State}
+import org.apache.james.jmap.json.{MailboxSerializer, ResponseSerializer}
+import org.apache.james.jmap.mail.{HasMoreChanges, MailboxChangesRequest, MailboxChangesResponse}
+import org.apache.james.jmap.routes.SessionSupplier
+import org.apache.james.mailbox.MailboxSession
+import org.apache.james.metrics.api.MetricFactory
+import play.api.libs.json.{JsError, JsSuccess}
+import reactor.core.scala.publisher.SMono
+
+import scala.jdk.CollectionConverters._
+import scala.jdk.OptionConverters._
+
+class MailboxChangesMethod @Inject()(mailboxSerializer: MailboxSerializer,
+                                     val metricFactory: MetricFactory,
+                                     val sessionSupplier: SessionSupplier,
+                                     val mailboxChangeRepository: MailboxChangeRepository) extends MethodRequiringAccountId[MailboxChangesRequest] {
+  override val methodName: MethodName = MethodName("Mailbox/changes")
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_MAIL)
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier], invocation: InvocationWithContext, mailboxSession: MailboxSession, request: MailboxChangesRequest): SMono[InvocationWithContext] =
+    if (request.sinceState.equals(State.INSTANCE)) {
+      val response: MailboxChangesResponse = MailboxChangesResponse(
+        accountId = request.accountId,
+        oldState = State.INSTANCE,
+        newState = State.INSTANCE,
+        hasMoreChanges = HasMoreChanges(false),
+        updatedProperties = Some(Properties()),
+        created = Set(),
+        updated = Set(),
+        destroyed = Set())
+      SMono.just(InvocationWithContext(invocation = Invocation(
+        methodName = methodName,
+        arguments = Arguments(mailboxSerializer.serializeChanges(response)),
+        methodCallId = invocation.invocation.methodCallId
+      ), processingContext = invocation.processingContext))
+    } else {
+      SMono.fromPublisher(mailboxChangeRepository.getSinceState(JavaAccountId.fromString(request.accountId.id), MailboxChange.State.of(request.sinceState.value), request.maxChanged.toJava))
```

Comment:
What do we do when the sinceState is not known?

Hint: return `cannotCalculateChanges`